src/items/itemDialog.component.tsx

```diff
@@ -482,28 +512,69 @@ function ItemDialog(props: ItemDialogProps) {
     setFormErrorMessage(undefined);
   }, [parentSystemId]);
 
+  const hasSerialNumberErrors =
+    advancedSerialNumberOptions.quantity &&
+    !itemDetails.serial_number?.trim().includes('%s') &&
+    'Please use %s to specify the location you want to append the number to serial number';
+
+  const hasQuantityErrors = !advancedSerialNumberOptions.quantity?.trim()
+    ? ''
+    : isNaN(Number(advancedSerialNumberOptions.quantity))
+      ? 'Please enter a valid number'
+      : !Number.isInteger(Number(advancedSerialNumberOptions.quantity))
+        ? 'Quantity must be an integer'
+        : Number(advancedSerialNumberOptions.quantity) <= 1
+          ? 'Quantity must be greater than 1'
+          : Number(advancedSerialNumberOptions.quantity) >= 100
+            ? 'Quantity must be less than 100'
+            : '';
+
+  const hasStartingValueErrors =
+    !advancedSerialNumberOptions.quantity &&
+    !!advancedSerialNumberOptions.startingValue?.trim()
+      ? 'Please enter a quantity value'
+      : isNaN(Number(advancedSerialNumberOptions.startingValue))
+        ? 'Please enter a valid number'
+        : !Number.isInteger(Number(advancedSerialNumberOptions.startingValue))
+          ? 'Starting value must be an integer'
+          : Number(advancedSerialNumberOptions.startingValue) < 0
+            ? 'Starting value must be greater than or equal to 0'
+            : '';
+
   const isStepFailed = React.useCallback(
     (step: number) => {
       switch (step) {
         case 0:
-          return Object.values(hasDateErrors).some(
-            (value: boolean) => value === true
+          return (
+            // Date error
+            Object.values(hasDateErrors).some(
+              (value: boolean) => value === true
+            ) ||
+            !!hasSerialNumberErrors ||
```

Comment:
does the `!!` not just cancel out? Or does it do something different than a boolean not not

Reply:
Yeah, but the hasSerialNumberErrors is not type boolean using !! changes it to a boolean truthy